pandas/tseries/index.py

```diff
@@ -1581,10 +1590,14 @@ def insert(self, loc, item):
         """
 
         freq = None
+
+        if isinstance(item, np.datetime64):
+            # np.datetime64 doesn't have tz, once convert to Timestamp
+            item = Timestamp(item)
```

Comment:
Ah, needs change. This is for current ``_has_same_tz`` cannot get ``tz`` from ``np.datetime64``.

https://github.com/pydata/pandas/pull/11343/files#diff-23ecb29e7ceba52109a365e447400d2eR562

1. move the condition to ``_has_same_tz`` ? ( convert ``np.datetime64`` to ``Timestamp``).
2. If we keep 1594th line, 1598th line can be  ``if isinstance(item, datetime):``

Maybe option 1 looks better?